yangson/instance.py

```diff
@@ -365,6 +366,29 @@ def validate(self: "InstanceNode", scope: ValidationScope = ValidationScope.all,
         """
         self.schema_node._validate(self, scope, ctype)
 
+    def get_error_list(self: "InstanceNode", scope: ValidationScope = ValidationScope.all,
+                 ctype: ContentType = ContentType.config) -> list:
+        """Get a list of validation exception objects as a result of validating the instance node.
+
+        Args:
+            scope: Scope of the validation (syntax, semantics or all).
+            ctype: Receiver's content type.
+
+        Returns:
+            A list with all validation exception objects
+        """
+        errors = set()
+        ignore_children = [RpcActionNode, InputNode, OutputNode, AnyContentNode, AnydataNode, AnyxmlNode]
+        if type(self.schema_node) in ignore_children or not self._children():
+            try:
+                self.validate(scope, ctype)
+            except (YangTypeError, SchemaError, SemanticError) as validationError:
+                errors.add(validationError)
+        else:
+            for child_node in self._children():
```

Comment:
if a node has _children and it also has "when"/"must"/"unique" conditions associated with it, this will skip the condition check and go straight to its children, giving potential false negatives.

Reply:
Shouldn't the validation method take that into consideration? In theory you can navigate to any part of the instance data and validate from there

Reply:
right, self.validate() would evaluate the conditions and report error. Conceptually you can think of the validate() method to take two steps for an internal node:
1) check only this node itself. 
2) if no errors are found, recursively check the node's descendent nodes. 
I think what should be done here is:
1) check only this node itself, capture any error and add to list
2) recursively call get_error_list() for this node's descendent nodes.

I think you are missing step(1), so if there is something wrong with this internal node itself, and it is not in ignore_children and it has children nodes, you will skip checking this internal node itself, and go straight to step(2) to check its children nodes.

Reply:
Made changes accordingly, hope it's looking better now